mozilla/fx-private-relay
# MPP-3644: Add Mozilla accounts opt-out property to Profile and API

The [Metrics for Relying Parties - Opt-Opt section](https://mozilla.github.io/ecosystem-platform/relying-parties/reference/metrics-for-relying-parties#user-opt-out) says:

> Users may opt-out of metrics from the FxA settings page. If they do, FxA will not collect or store metrics and relying parties must not either.
>
> [User profiles](https://github.com/mozilla/fxa/blob/main/packages/fxa-profile-server/docs/API.md#get-v1profile) include the `metricsEnabled` boolean. When the value is `false` relying parties should not collect any metrics tied to the user. Relying parties should check this value every time the profile is requested.

For MPP-3644, this PR implements reading this value on the backend, and forwarding it via the API.

This also includes front-end changes to use this new signal to decide if we should load Google Analytics.

# Testing

1. Run `npm -w frontend install` to install the new `react-singleton-hook` dependency
2. Run `npm -w frontend run watch` to build the frontend with debugging on
3. Start up the server locally with `./manage.py runserver`,
4. Load http://127.0.0.1:8000/ , open developer tools, and reload
    - [x] You should see Logs in the dev tools Console about Google Analytics loading, sending events and pageviews
5. If not logged in, log in
    - [x] You should see Logs in the dev tools Console about Google Analytics loading, sending events and pageviews
6. Navigate to http://127.0.0.1:8000/api/v1/profiles/
    - [x] You should see `"metrics_enabled": true` in the JSON response.
7. Go to https://accounts.stage.mozaws.net, log in if needed, and deselect "Help improve ⁨Mozilla accounts⁩" under "Data Collection and Use".
8. Go to http://127.0.0.1:8000/accounts/logout/ to log out, ensure the dev tools are open, clear the Console output, and then log in again
   - [x] After log in, Google Analytics is not loaded.
9. Go to http://127.0.0.1:8000/api/v1/profiles/.
  - [x] You should see `"metrics_enabled": false`
